### DISCUSSION TITLE
Pause a queue

### DISCUSSION THREAD

**Discussion by lpoiana** [Templum/rabbitmq-connector#183] [Q&A]
Is there the possibility to pause a queue without losing messages?
Ps: i'm using durable=True on exchange declaration and also properties=pika.BasicProperties(delivery_mode=2) on publish.

Thanks in advance

**Comment by Templum** [comment]
Not so sure how you even imagine a pause of the queue. The service itself will always read, so one thing would be to stop writing to the queue.

The other would be to scale down the service, stopping it from reading from the queue.

**Reply by lpoiana** [reply]
Consider the following scenario: 
Users are writing to the queue, and we need to put the backend(receive functions) into maintenance. The payloads sent by the users needs to stay in the queue and as soon as the backend is online again it must receive the payloads.
Already tried the option to stop the service, but i don't have the desired outcome even if i'm using the parameters mentioned in the first post.
Any idea on how can i achive the scenario mentioned?
Your response is much appreciated.
Thak you

**Reply by Templum** [reply]
Well, I'm not a Rabbit MQ Expert and this question seems to me as it would be not specific to my connector. So there is no real advice I can offer you if stopping the service (connector) is not working for you. 

Another solution would be to remove relevant topics from the function, however here you would need to consider the refresh time. And then later add it back.

**Reply by lpoiana** [reply]
Just FYI, i was able have the desired behavior by applying the durable=true also to the queue in the send function
